src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -261,6 +263,10 @@ Object getVectorValueAt(String column, int index) {
     if (logicalType == ColumnLogicalType.BINARY && value != null) {
       value = value instanceof String ? ((String) value).getBytes(StandardCharsets.UTF_8) : value;
     }
+    /* Mismatch between Iceberg string & FDN String */
+    if (Objects.equals(columnMetadata.getSourceIcebergDataType(), "\"string\"")) {
+      value = value instanceof byte[] ? new String((byte[]) value, StandardCharsets.UTF_8) : value;
```

Comment:
this is test-only code so np, but otherwise assuming the byte[] is UTF8 is a bit risky and shouldn't be done.